astropy/io/ascii/tests/test_tdat.py

```diff
@@ -0,0 +1,394 @@
+        t.write(out, format="ascii.tdat")
+
+    t.meta["table_name"] = "anastropytablewithanamethatistoolongfortheformat"
+    with pytest.warns(TdatFormatWarning, match="'table_name' is too long"):
+        t.write(out, format="ascii.tdat")
+
+    t.meta["table_name"] = "astropy_table"
+    t.meta["table_description"] = """
+    This is a description that exceeds the character limit allowed by the tdat
+    format and it should be truncated before being written. A warning should
+    pop up to inform the user of this behavior.
+    """
+    with pytest.warns(TdatFormatWarning, match="'table_description' is too long"):
+        t.write(out, format="ascii.tdat")
+
+
+def test_write_simple():
+    """
+    Write a simple table with common types.  This shows the compact version
+    of serialization with one line per column.
+    """
+    with warnings.catch_warnings():
+        warnings.filterwarnings("ignore", category=TdatFormatWarning)
+        t = simple_table()
+        out = StringIO()
+        t.write(out, format="ascii.tdat")
+        assert out.getvalue().splitlines() == SIMPLE_LINES
+
+
+def test_write_full():
+    """
+    Write a full-featured table with common types and explicitly check output
+    """
+    t = test_table
+    lines = [
+        "<HEADER>",
+        "# TABLE: heasarc_simple",
+        "# TOTAL ROWS: 7",
+        "table_name = heasarc_simple",
+        'table_description = "Test table"',
+        "table_security = public",
+        "#",
+        "# Table Parameters",
+        "#",
+        "field[record_number] = int4 [meta.id] (key) // Unique Identifier for Entry",
+        "field[id] = int4 [meta.id] (index) // Source ID Number",
+        "field[name] = char3 [meta.id;meta.main] (index) // String Name",
+        "field[ra] = float8:.4f_deg [pos.eq.ra] (index) // Right Ascension",
+        "field[dec] = float8:.4f_deg [pos.eq.dec] (index) // Declination",
+        "field[empty] = float8:.4f // Empty // Comment",
+        "#",
+        "parameter_defaults = name ra dec",
+        "#",
+        "# Virtual Parameters",
+        "#",
+        "frequency_regime = Gamma-ray",
+        "observatory_name = GAMMA-RAY BURSTS",
+        "row_type = GRB",
+        "table_author = Example et al.",
+        "table_priority = 3.01",
+        "table_type = Observation",
+        "unique_key = record_number",
+        "#",
+        "# Data Format Specification",
+        "#",
+        "line[1] = record_number id name ra dec empty",
+        "#",
+        "<DATA>",
+        "1|10|aaa|1.0000|1.0000||",
+        "2|20|b|2.0000|||",
+        "3|30|c||3.0000||",
+        "4|20|||||",
+        "5||||||",
+        "|60|f|6.0000|6.0000||",
+        "7|70|g|7.0000|7.0000||",
+        "<END>",
+    ]
+
+    out = StringIO()
+    t.write(out, format="ascii.tdat")
+    assert out.getvalue().splitlines() == lines
+
+
+def test_write_read_roundtrip():
+    """
+    Write a full-featured table with all types and see that it round-trips on
```

Comment:
The test `test_write_full` tests that TDAT -> Table -> TDAT round-trips. 

This test is doing the converse of Table -> TDAT -> Table. That's good, but there are no tests anywhere that the Table itself looks as expected. Are the Table columns all correct (dtype, description, units, etc)?  Is the Table meta correct? Just because it round-trips to the same TDAT file doesn't necessarily imply that the Table value is correct.

An example is that the `'table_description = "Test table"'` header key/value is being parsed as the string `'"Test table"'` with the surrounding double quotes in the string value. By not confirming the values of the Table meta, issues like this can be missed.